Remove non-standard encodings (e.g., 'x-mac-turkish', 'x-mac-greek' etc.)

Ahmad-S792 opened the pull request:
#### a78cd83130cb3ea242762121be17a905f201b154
<pre>
Remove non-standard encodings (e.g., &apos;x-mac-turkish&apos;, &apos;x-mac-greek&apos; etc.)

<a href="https://bugs.webkit.org/show_bug.cgi?id=265261">https://bugs.webkit.org/show_bug.cgi?id=265261</a>
<a href="https://rdar.apple.com/problem/118944539">rdar://problem/118944539</a>

Reviewed by Alex Christensen.

This patch aligns WebKit with Gecko / Firefox and Blink / Chromium by removing following
non-standard encodings:

- x-mac-ce
- x-mac-centraleurroman
- x-mac-turkish
- x-mac-greek
- euc-tw
- windows-936-2000

* Source/WebCore/PAL/pal/text/TextCodecCJK.cpp:
(TextCodecCJK::registerEncodingNames):
* Source/WebCore/PAL/pal/text/TextCodecICU.cpp:
(DECLARE_ALIASES):
(EncodingName):
* LayoutTests/imported/w3c/web-platform-tests/encoding/unsupported-labels-expected.txt: Rebaselined
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/encoding/unsupported-labels-expected.txt: Rebaselined
* LayoutTests/fast/encoding/char-decoding-mac.html: Rebaselined
* LayoutTests/fast/encoding/char-decoding-mac-expected.txt: Rebaselined
* LayoutTests/fast/encoding/legacy-tec-encodings.html: Rebaselined
* LayoutTests/fast/encoding/legacy-tec-encodings-expected.txt: Rebaselined
* LayoutTests/fast/encoding/char-encoding-mac.html: Rebaselined
* LayoutTests/fast/encoding/char-encoding-mac-expected.txt: Rebaselined
* LayoutTests/fast/encoding/legacy-ios-encodings.html: Rebaselined
* LayoutTests/fast/encoding/legacy-ios-encodings-expected.txt: Rebaselined

Canonical link: <a href="https://commits.webkit.org/275797@main">https://commits.webkit.org/275797@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/dea4685140fbd1f140a5882381852ea91f01bf32

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/42646 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/21668 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/45048 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/45256 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/38770 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/44953 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/25427 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/19033 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/35298 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/43220 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/18775 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/36707 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/16249 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/16404 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/37766 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/703 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/38918 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/38095 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/46752 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/17465 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/14391 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/42004 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/19084 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/37035 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/40620 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/19263 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5804 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/18729 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/f9eff846ba01472ba0b9df0132698d99a90da449. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/22273

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/78c63b78feec2208cf5413d911bba18cddd5103a. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/22273

Ahmad-S792 commented:
NOTE - I am in favour of WPT tests and deleting local tests. Although I will wait for input from other developers.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/6957b6e89d6109038661bfc630e66a2287307fee. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/22273

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/134802195fe75f0afd5ae1f7456e2b83fb1f8fe8. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/22273

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/dea4685140fbd1f140a5882381852ea91f01bf32. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/22273

achristensen07 commented:
This is a difficult one.  I understand that other browsers don't support those encodings, but is there a reason for us to remove support for those encodings?  What kind of content uses or used those encodings, and what has happened to that content?

Ahmad-S792 commented:
> This is a difficult one. I understand that other browsers don't support those encodings, but is there a reason for us to remove support for those encodings? What kind of content uses or used those encodings, and what has happened to that content?

My main inspirations to do this PR are:

1) Align with other browsers and fix failing WPT test cases
2) It is not in Web Standard - https://encoding.spec.whatwg.org/#names-and-labels
3) To try to achieve (https://bugs.webkit.org/show_bug.cgi?id=179303#c7):

_"I think we should eliminate as many encoding names as we can, and synchronize with the encoding specification. Any encoding names that we decide to continue to support that is not mentioned in the specification needs a really good rationale; perhaps such encoding names can be limited to the context where they are required or, alternatively, added to the encoding specification and to other web browser engines."_

Ahmad-S792 committed:
Remove non-standard encodings (e.g., 'x-mac-turkish', 'x-mac-greek' etc.)

https://bugs.webkit.org/show_bug.cgi?id=265261
rdar://problem/118944539

Reviewed by Alex Christensen.

This patch aligns WebKit with Gecko / Firefox and Blink / Chromium by removing following
non-standard encodings:

- x-mac-ce
- x-mac-centraleurroman
- x-mac-turkish
- x-mac-greek
- euc-tw
- windows-936-2000

* Source/WebCore/PAL/pal/text/TextCodecCJK.cpp:
(TextCodecCJK::registerEncodingNames):
* Source/WebCore/PAL/pal/text/TextCodecICU.cpp:
(DECLARE_ALIASES):
(EncodingName):
* LayoutTests/imported/w3c/web-platform-tests/encoding/unsupported-labels-expected.txt: Rebaselined
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/encoding/unsupported-labels-expected.txt: Rebaselined
* LayoutTests/fast/encoding/char-decoding-mac.html: Rebaselined
* LayoutTests/fast/encoding/char-decoding-mac-expected.txt: Rebaselined
* LayoutTests/fast/encoding/legacy-tec-encodings.html: Rebaselined
* LayoutTests/fast/encoding/legacy-tec-encodings-expected.txt: Rebaselined
* LayoutTests/fast/encoding/char-encoding-mac.html: Rebaselined
* LayoutTests/fast/encoding/char-encoding-mac-expected.txt: Rebaselined
* LayoutTests/fast/encoding/legacy-ios-encodings.html: Rebaselined
* LayoutTests/fast/encoding/legacy-ios-encodings-expected.txt: Rebaselined

Canonical link: https://commits.webkit.org/275797@main

webkit-commit-queue commented:
Committed 275797@main (a78cd83130cb): <https://commits.webkit.org/275797@main>

Reviewed commits have been landed. Closing PR #22273 and removing active labels.
